pkg/object/object.go

```diff
@@ -45,9 +48,27 @@ type OpenCompose struct {
 	Volumes  []Volume
 }
 
+func (s *Service) Validate() error {
+	// validate service name, like it cannot have underscores, etc.
+
+	// validate containers
+
+	// validate replicas
+	if s.Replicas != nil && *s.Replicas < 0 {
```

Comment:
Logically, the value is valid for parser because parser is only supposed to parse data for its type and not validate if it is correct or not. That's the job of validator.

And I would like to have validation happening in only one place not twice. Someone using opencompose as lib might not be passing validated data so how are we gonna validate that?